mailpoet/lib/Migrations/App/Migration_20240704_104122_App.php

```diff
@@ -0,0 +1,62 @@
+<?php declare(strict_types = 1);
+
+namespace MailPoet\Migrations\App;
+
+use MailPoet\Entities\FormEntity;
+use MailPoet\Migrator\AppMigration;
+use MailPoet\Util\Helpers;
+
+class Migration_20240704_104122_App extends AppMigration {
+  public function run(): void {
+    $this->replacePrivacyPolicyLinkTags();
+  }
+
+  private function replacePrivacyPolicyLinkTags(): void {
+    $formsTable = $this->getTableName(FormEntity::class);
+
+    // Only privacy policy links had [link] tags in stock templates
+    $qb = $this->entityManager->createQueryBuilder();
+    $formsWithLinkTags = $qb->select('f')
+      ->from(FormEntity::class, 'f')
+      ->where($qb->expr()->like('f.body', $qb->expr()->literal('%[link]%')))
+      ->orderBy('f.id')
+      ->getQuery()
+      ->getResult();
+
+    foreach ($formsWithLinkTags as $form) {
+      $this->processFormLinkTags($form);
+    }
+    $this->entityManager->flush();
+  }
+
+  private function processFormLinkTags(FormEntity $form) {
+    $body = $form->getBody();
+    if (!is_array($body)) { // Sanity check
+      return;
+    }
+    $changed = false;
+    foreach ($body as $key => $block) {
+      // Skip blocks of irrelevant types and malformed blocks
+      if (!isset($block['type']) || $block['type'] !== 'paragraph' || !isset($block['params']['content'])) {
+        continue;
+      }
+      // Skip blocks without link tags
+      if (!strpos($block['params']['content'], '[link]')) {
+        continue;
+      }
+      $block['params']['content'] = Helpers::replaceLinkTags($block['params']['content'], '#');
```

Comment:
Shouldn't we use `get_privacy_policy_url` to get the proper URL instead of replacing it with `#`? This way we remove `[link]` from the form, but will create a link that doesn't go to the privacy policy page. So actually this won't help much...

Thinking about it, I would hope that users would notice that a form they placed on their site contains a strange [link] and replace it with the correct link or remove it. So maybe we don't need the migration. What do you think?

Reply:
> Shouldn't we use get_privacy_policy_url to get the proper URL instead of replacing it with #?

Fair point. As I mentioned in the description, we can't tell privacy policy link tags from other link tags, so I was a bit concerned that we can encounter unrelated link tags in the wild, and having a dummy link is better than a wrong link in this case, in my opinion.

> Thinking about it, I would hope that users would notice that a form they placed on their site contains a strange [link] and replace it with the correct link or remove it. So maybe we don't need the migration. What do you think?

Since we haven't encountered many complaints about this, and this isn't a high-impact issue if these tags are kept in old forms, we can remove this migration.